tests/testthat/test-text.R

```diff
@@ -18,6 +18,17 @@ test_that("renderPrint and renderText behavior is correct", {
                'foo')
   expect_equal(isolate(renderText({ invisible("foo") })()),
                'foo')
+  expect_equal(isolate(renderText({ c("hello", "world") })()),
+               'hello world')
+
+  expect_equal(isolate(renderText({ c("hello", "world") }, sep=", ")()),
+               'hello, world')
+  # Really just measuring the behavior of `cat` when it receives a vector of
+  # separators, but probably worth covering.
+  expect_equal(isolate(renderText({ c("hello", "there", "world", "repeats") },
```

Comment:
Tbh I think you should make this an error rather than allowing very loose base behaviour to flow upwards.